src/upx.mk

```diff
@@ -47,7 +47,9 @@ define $(PKG)_BUILD_$(BUILD)
         'CC=$(BUILD_CC)' \
         'PKG_CONFIG=$(PREFIX)/$(BUILD)/bin/pkgconf' \
         'LIBS=-L$(PREFIX)/$(BUILD)/lib -lucl -lz' \
-        $(shell [ `uname -s` == Darwin ] && echo "CXXFLAGS='-Wno-error=unused-local-typedef'") \
+        $(shell [ `uname -s` == Darwin ] && \
+            echo "CXXFLAGS=-Wno-error=unused-local-typedefs -Wno-error=misleading-indentation" || \
+            echo "CXXFLAGS=-Wno-error=misleading-indentation") \
```

Comment:
I see now, couldn't actually see either the original error with gcc6 on Darwin as setting `CXXFLAGS` to anything disables the `-Werror` set by upx. The second option `-Wno-error=misleading-indentation` isn't quoted correctly and is passed an option to `make` instead of `CXXFLAGS` so didn't affect gcc5.

I'm pretty sure [this fix](https://github.com/mxe/mxe/commit/d21c1455d3b6788f2b3340a49b29d12832e50688) (along with some other related changes in #1496) will work across all platforms.